src/Connection/Connection.php

```diff
@@ -320,12 +291,12 @@ private function validateResponse(array $response, int $responseToken, int $toke
             );
         }
 
-        if ($response['t'] === ResponseType::COMPILE_ERROR) {
-            throw new ConnectionException('Compile error: ' . $response['r'][0] . ', jsonQuery: ' . json_encode($message));
+        if ($response->getType() === ResponseType::COMPILE_ERROR) {
```

Comment:
Will more cases be added? Since I can see a lot of IF statements coming which basically do the same

Reply:
No, this will be it for now.